src/xbot/common/math/PID.java

```diff
@@ -102,17 +142,24 @@ public double calculate(double goal, double current,
             double p, double i, double d) {
         return calculate(goal, current, p, i, d, 0);
     }
-
+    
     /**
-     * This tells you if the controller has met its goal. Don't call this before
-     * calling calculate!
-     * 
-     * @param absoluteTolerance
-     *            How close the value can be before it is considered
-     *            "on-target."
+     * Check for all conditions where the tolerance is above 0. Since tolerances default to
+     * -1, it will skip any unassigned ones.
      */
-    public boolean isOnTarget(double absoluteTolerance)
-    {
-        return Math.abs(m_targetInputValue - m_currentInputValue) < absoluteTolerance;
+    public boolean isOnTarget() {
+        boolean checkError = errorTolerance > 0;
+        boolean checkDerivative = derivativeTolerance > 0;
+        
+        boolean isOnTarget = true;
```

Comment:
If you're calling isOnTarget and you didn't set a tolerance, we can do two simple things:
-Always say true. Command ends immediately.
-Always say false. Command never ends.

Both of these are bad - but by having the command immediately end, we actually highlight the problem faster. If you were doing tests of a command on the robot, we could notice quickly that the command was starting but stopping immediately, which would let us to investigate isFinished.

There's probably some more advanced stuff we could do to try and highlight this to the programmer (maybe a test-only exception, and/or error logging), but for what it does on the robot I think we have to choose one of the two approaches above.

Do you have any feedback on which approach is more desirable?